### ISSUE TITLE
Unable to initialize gateway backend: unable to initialize hdfsClient

### ISSUE THREAD

**Issue by lynch2git** [minio/minio#13210]
Hello everyone, I encountered a issue.

## Expected Behavior

I am trying to start a minio using gateway hdfs with kerberos.

## Current Behavior

When i starting minio, it report an error.

## Steps to Reproduce (for bugs)
1. init a shell with all the needed:
export MINIO_ACCESS_KEY=minio
export MINIO_SECRET_KEY=minio123
export HADOOP_CONF_DIR=/opt/hdfsclient/Hadoop/hadoop/conf
export HADOOP_HOME=/opt/hdfsclient/Hadoop/hadoop/hadoop
export KRB5_CONFIG=/opt/apps/minio/krb5.conf
export KRB5KEYTAB=/opt/apps/minio/user1.keytab
export KRB5USERNAME=user1
export KRB5REALM=DATA02.COM
./minio gateway hdfs hdfs://192.168.1.2:8020

2. exec the shell
[root@dfs-3 run]# ./run.sh

3. retrieve error :
ERROR Unable to initialize gateway backend: unable to initialize hdfsClient

## Context

1. There is no problem executing the command 'ping 192.168.1.2' and 'telnet 192.168.1.2 8020' on my minio node.

2. Executing the command `klist` on my minio node:
```
[root@dfs-3 minio]# klist
Ticket cache: FILE:/tmp/krb5cc_0
Default principal: user1@DATA02.COM

Valid starting       Expires              Service principal
04/25/2021 18:35:27  03/30/2026 18:35:26  krbtgt/DATA02.COM@DATA02.COM
```

3. Executing the command `hdfs dfs -ls` on my minio node is ok:
```
[root@dfs-3 minio]# hdfs dfs -ls hdfs://192.168.1.2:8020/
Found 10 items
drwxr-xr-x   - hdfs   hdfs            0 2021-04-20 16:49 hdfs://192.168.1.2:8020/apps
drwx------   - hdfs   hdfs            0 2021-04-21 17:03 hdfs://192.168.1.2:8020/flink
drwxr-xr-x   - hdfs   hdfs            0 2021-04-20 16:43 hdfs://192.168.1.2:8020/hdp
drwxr-xr-x   - mapred hdfs            0 2021-04-20 16:43 hdfs://192.168.1.2:8020/mapred
drwxrwxrwx   - mapred hadoop          0 2021-04-20 16:44 hdfs://192.168.1.2:8020/mr-history
drwxr-xr-x   - hdfs   hdfs            0 2021-04-20 16:44 hdfs://192.168.1.2:8020/services
drwxrwxrwx   - spark  hadoop          0 2021-04-23 17:23 hdfs://192.168.1.2:8020/spark2-history
drwx------   - hdfs   hdfs            0 2021-04-21 18:33 hdfs://192.168.1.2:8020/tenant
drwxrwxrwx   - hdfs   hdfs            0 2021-04-22 11:30 hdfs://192.168.1.2:8020/tmp
drwxrwxrwx   - hdfs   hdfs            0 2021-04-21 17:02 hdfs://192.168.1.2:8020/user
```

4. The namenode has no error log when i starting minio.

## Your Environment

* Version used (`minio --version`):
minio version RELEASE.2021-04-22T15-44-28Z

* HDFS Version:
3.0.0

* Server setup and configuration:
SELinux and IPTables from the environment are disabled.

* Operating System and version (`uname -a`):
Linux dfs-3 3.10.0-514.el7.x86_64 1 SMP Tue Nov 22 16:42:41 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux

* This is part of the my HDFS setup and minio setup:
[mysetup.zip](https://github.com/minio/minio/files/7166472/mysetup.zip)

Anyone encountered this issue ?

Regards,
Lynch

**Comment by harshavardhana** [comment]
> * Version used (`minio --version`):
>   minio version RELEASE.2021-04-22T15-44-28Z

Please use latest release @lynch2git

**Comment by lynch2git** [comment]
I tried latest release, it start failed and has no prompt information, like this: 
```
[root@dfs-3 hdfs]# export ... (some variables)
[root@dfs-3 hdfs]# ./minio gateway hdfs hdfs://192.168.1.2:8020
[root@dfs-3 hdfs]# 
```
Version used (`minio --version`):
minio version RELEASE.2021-09-09T21-37-07Z

@harshavardhana

**Comment by lynch2git** [comment]
How to get more detailed error log ?

**Comment by harshavardhana** [comment]
@lynch2git - please upgrade we print the actual error as well. 

```
~ ~/hdfs.sh
ERROR Unable to initialize gateway backend: unable to initialize hdfsClient: no available namenodes: dial tcp 192.168.1.2:8020: i/o timeout
```

**Comment by lynch2git** [comment]
Thank you, I have found the cause of the problem: my standby namenode is down, although the active namenode is ok.
